src/controller/python/chip/yaml/YamlParser.py

```diff
@@ -0,0 +1,351 @@
+
+        test.expected_response_data = YamlUtils.fixup_yaml_types(
+            test.expected_response_data, test.expected_response_object)
+        test.expected_response_object = test.expected_response_object.FromDict(
+            test.expected_response_data)
+
+    return test
+
+
+def AttributeReadCommandFactory(item: dict, cluster: str):
+    '''Parse read attribute command from yaml test configuration.
+
+    Args:
+      'item': Dictionary contains single read attribute test to be parsed
+      'cluster': Name of cluster read attribute command is targeting.
+    Returns:
+      ReadAttributeCommand if 'item' is a valid command to be executed.
+      None if 'item' was not parsed to a command to be executed for a known reason
+        that is none fatal.
+    Raises:
+      ParsingError: Raised if there is an unexpected parsing error.
+    '''
+    test = ReadAttributeCommand(item['label'], stringcase.pascalcase(item['attribute']), cluster)
+
+    try:
+        test.request_object = eval(f'Clusters.{test.cluster}.Attributes.{test.request_type_name}')
+    except AttributeError:
+        pass
+
+    if test.request_object is None:
+        raise ParsingError(
+            f'ReadAttribute failed to find cluster:{test.cluster} '
+            f'Attribute:{test.request_type_name}')
+
+    if (item.get('arguments')):
+        raise ParsingError(f'ReadAttribute should not contain arguments. {test.label}')
+
+    if test.request_object.attribute_type is None:
+        raise ParsingError(f'ReadAttribute doesnt have valid attribute_type. {test.label}')
+
+    test.expected_raw_response = item.get('response')
+    if (test.expected_raw_response is None):
+        raise ParsingError(f'ReadAttribute missing expected response. {test.label}')
+
+    if 'optional' in item:
+        test.possibly_unsupported = True
+
+    if 'value' in test.expected_raw_response:
+        test.expected_response_object = test.request_object.attribute_type.Type
+        value = test.expected_raw_response['value']
+        test.expected_response_data = YamlUtils.fixup_yaml_types(
+            value, test.expected_response_object, use_from_dict=True)
+
+    return test
+
+
+def AttributeWriteCommandFactory(item: dict, cluster: str):
+    '''Parse write attribute command from yaml test configuration.
+
+    Args:
+      'item': Dictionary contains single write attribute test to be parsed
+      'cluster': Name of cluster write attribute command is targeting.
+    Returns:
+      WriteAttributeCommand if 'item' is a valid command to be executed.
+      None if 'item' was not parsed to a command to be executed for a known reason
+        that is none fatal.
+    Raises:
+      ParsingError: Raised if there is an unexpected parsing error.
+    '''
+    test = WriteAttributeCommand(item['label'], stringcase.pascalcase(item['attribute']), cluster)
+
+    try:
+        test.request_object = eval(f'Clusters.{test.cluster}.Attributes.{test.request_type_name}')
```

Comment:
This can cause arbitrary code execution. This should be replaced with walking the metadata tree of the class hierarchies involved.